sdk/core/src/pipeline.rs

```diff
@@ -79,8 +80,36 @@ where
         // TODO: Add transport policy for WASM once https://github.com/Azure/azure-sdk-for-rust/issues/293 is resolved.
         #[cfg(not(target_arch = "wasm32"))]
         {
-            let transport_policy = TransportPolicy::new(&options.transport);
-            pipeline.push(Arc::new(transport_policy));
+            // This code replaces the default transport policy at runtime if these two conditions
+            // are met:
+            // 1. The mock_transport_framework is enabled
+            // 2. The environmental variable TESTING_MODE is either RECORD or PLAY
+            #[cfg(not(feature = "mock_transport_framework"))]
+            pipeline.push(Arc::new(TransportPolicy::new(options.transport.clone())));
+
+            #[cfg(feature = "mock_transport_framework")]
+            match &std::env::vars()
+                .find(|env| env.0 == "TESTING_MODE")
+                .unwrap_or_default()
+                .1
+                .to_uppercase() as &str
+            {
+                "RECORD" => {
+                    warn!("mock testing framework record mode enabled");
+                    pipeline.push(Arc::new(crate::policies::MockTransportRecorderPolicy::new(
+                        options.transport.clone(),
+                    )));
+                }
+                "PLAY" => {
+                    warn!("mock testing framework record reply enabled");
```

Comment:
nit: warn seems like a strange choice. Why not debug or info?

Reply:
I went with `warn` because changing the behavior of the transmission policy to record/reply Azure interactions has security implications. 
It's not a warning in the sense of "the code behaved unexpectedly", it's more of a "hey, watch out, the normal behavior has been changed, possibly giving forged responses to your queries".

Do you think we should come up with another way to convey this information?